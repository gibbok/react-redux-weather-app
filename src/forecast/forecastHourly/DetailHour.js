// @flow
import * as React from 'react'
import moment from 'moment'
import IconWeather from '../../shared/icon/IconWeather'
import IconWind from '../../shared/icon/IconWind'
import { withStyles, Typography, Divider } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +momentId: string,
  +timespan: number,
  +temp: number,
  +weatherIconCode: number,
  +weatherDescription: string,
  +windDegree: number,
  +windSpeed: number,
  +humidityUnit: '%',
  +humidityValue: number,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
      width: 112
    },
    icon: {
      marginBottom: spaceUnit * 0.5
    },
    temperature: {
      marginBottom: spaceUnit * 0.5
    },
    description: {
      marginBottom: spaceUnit
    },
    wind: {},
    humidity: {
      marginBottom: spaceUnit * 1.25
    },
    hour: {}
  })
}

const DetailHour:React.StatelessFunctionalComponent<PropsType> = ({ momentId, timespan, temp, weatherIconCode, weatherDescription, windDegree, windSpeed, humidityUnit, humidityValue, classes }:PropsType):React.Element<any> => {
  const hourFormat = moment.unix(timespan).format('h A')
  const tempFormat = Math.round(temp)
  const windSpeedFormat = Math.round(windSpeed)
  return (
    <div className={classes.root}>
      <div className={classes.icon}>
        <Typography type='headline'>
          <IconWeather code={weatherIconCode} />
        </Typography>
      </div>
      <div className={classes.temperature}>
        <Typography type='title'>
          {tempFormat} &#176;
        </Typography>
      </div>
      <div className={classes.description}>
        <Typography type='subheading'>
          {weatherDescription}
        </Typography>
      </div>
      <div className={classes.wind}>
        <Typography type='body1'>
          Wind <IconWind degree={windDegree} /> {windSpeedFormat} mps
        </Typography>
      </div>
      <div className={classes.humidity}>
        <Typography type='body1'>
          Humidity {humidityValue} {humidityUnit}
        </Typography>
      </div>
      <div className={classes.hour}>
        <Divider />
        <Typography type='subheading'>
          <strong>{hourFormat}</strong>
        </Typography>
      </div>
    </div>
  )
}

export default withStyles(styles)(DetailHour)
