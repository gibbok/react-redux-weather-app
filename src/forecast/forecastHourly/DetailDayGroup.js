// @flow
import * as React from 'react'
import moment from 'moment'
import DetailHour from './DetailHour'
import { withStyles, Typography, Divider } from 'material-ui'

/* eslint-disable no-undef */
type PropsType = {
  +timespan:number,
  +hours: Array<{
    +humidityUnit: '%',
    +humidityValue: number,
    +momentId: string,
    +temp: number,
    +timespan: number,
    +weatherDescription: string,
    +weatherIconCode: number,
    +windDegree: number,
    +windSpeed: number
  }>,
  +classes: Object
}
/* eslint-enable no-undef */

const styles = theme => {
  const spaceUnit = theme.spacing.unit
  return ({
    root: {
    },
    title: {
      marginBottom: spaceUnit * 1
    },
    hours: {
      display: 'flex',
      marginTop: spaceUnit * 1.5,
      marginBottom: spaceUnit * 3
      // borderTop: '1px solid rgba(0,0,0,0.5)',
      // borderBottom: '1px solid rgba(0,0,0,0.5)'
    }
  })
}

const DetailDayGroup:React.StatelessFunctionalComponent<PropsType> = ({ timespan, hours, classes }:PropsType):React.Element<any> => {
  const dayFormat = moment.unix(timespan).format('ddd, MMM D')
  const hoursDetails = hours.map((day, index) =>
    <DetailHour
      key={index}
      {...hours[index]}
    />
  )
  return (
    <div className={classes.root}>
      <div className={classes.title}>
        <Typography type='title'>
          {dayFormat}
        </Typography>
      </div>
      <Divider />
      <div className={classes.hours}>
        {hoursDetails}
      </div>
    </div>
  )
}

export default withStyles(styles)(DetailDayGroup)
