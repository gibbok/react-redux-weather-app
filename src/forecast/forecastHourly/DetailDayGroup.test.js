// @flow
import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import DetailDayGroup from './DetailDayGroup'
import DetailHour from './DetailHour'

describe('<DetailDayGroup />', () => {
  it('should render', () => {
    const data = {
      timespan: 1505206800,
      hours: [
        {
          humidityUnit: '%',
          humidityValue: 93,
          momentId: '2017:8:12:2',
          temp: 14.42,
          timespan: 1505206800,
          weatherDescription: 'Clouds',
          weatherIconCode: 803,
          windDegree: 273.006,
          windSpeed: 1.26
        }
      ]
    }
    const wrapper = shallow(
      <DetailDayGroup
        timespan={data.timespan}
        hours={data.hours}
      />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should render one DetailHour component', () => {
    const data = {
      timespan: 1505206800,
      hours: [
        {
          humidityUnit: '%',
          humidityValue: 93,
          momentId: '2017:8:12:2',
          temp: 14.42,
          timespan: 1505206800,
          weatherDescription: 'Clouds',
          weatherIconCode: 803,
          windDegree: 273.006,
          windSpeed: 1.26
        }
      ]
    }
    const wrapper = shallow(
      <DetailDayGroup
        timespan={data.timespan}
        hours={data.hours}
        />
      )
    expect(wrapper.find(DetailHour).length).toBe(1)
  })
})
