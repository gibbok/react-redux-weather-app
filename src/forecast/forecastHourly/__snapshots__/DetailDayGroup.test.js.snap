// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailDayGroup /> should render 1`] = `
<div
  className="DetailDayGroup-root-1"
>
  <div
    className="DetailDayGroup-title-2"
  >
    <withStyles(Typography)
      type="title"
    >
      Tue, Sep 12
    </withStyles(Typography)>
  </div>
  <withStyles(Divider) />
  <div
    className="DetailDayGroup-hours-3"
  >
    <withStyles(DetailHour)
      humidityUnit="%"
      humidityValue={93}
      momentId="2017:8:12:2"
      temp={14.42}
      timespan={1505206800}
      weatherDescription="Clouds"
      weatherIconCode={803}
      windDegree={273.006}
      windSpeed={1.26}
    />
  </div>
</div>
`;
